test/test_nn.py

```diff
@@ -4515,6 +4515,126 @@ def test_spectral_norm_pickle(self):
         m = pickle.loads(pickle.dumps(m))
         self.assertIsInstance(m, nn.Linear)
 
+    def test_orthogonal_parametrization(self):
+        # Orthogonal implements 6 algorithms (3x parametrizations times 2 options of extra_memory)
+
+        def assert_is_orthogonal(X):
+            n, k = X.size(-2), X.size(-1)
+            if n < k:
+                X = X.transpose(-2, -1)
+                n, k = k, n
+            Id = torch.eye(k, dtype=X.dtype, device=X.device).expand(*(X.size()[:-2]), k, k)
+            eps = 10 * n * torch.finfo(X.dtype).eps
+            torch.testing.assert_allclose(X.transpose(-2, -1).conj() @ X, Id, atol=eps, rtol=0.)
+
+        for shape, dtype, batched, in product(((4, 4), (5, 3), (3, 5)),  # square/ tall / wide
+                                              (torch.float32, torch.complex64),
+                                              (False, True)):
+            # Conv2d does not support complex yet
+            if batched and dtype.is_complex:
+                continue
+
+            if batched:
+                input = torch.randn(2, 2, shape[0] + 2, shape[1] + 1, dtype=dtype)
+            else:
+                input = torch.randn(3, shape[0], dtype=dtype)
+
+            for parametrization, extra_memory in product(("matrix_exp", "cayley", "householder"),
+                                                         (False, True)):
+                # right_inverse for Cayley and matrix_exp not implemented for extra_memory=False
+                # See Note [right_inverse expm cayley]
+                can_initialize = extra_memory or parametrization == "householder"
+
+                # We generate them every time to always start with fresh weights
+                if batched:
```

Comment:
I'm not sure why batched means conv vs linear here?

Reply:
The weight of the convolution is considered as a batch of filters. I just needed a layer that has a tensor inside (`ndim > 2`) and a layer that has a matrix inside to test that it works as expected in both cases.

Reply:
I would just call it use_conv/use_linear then?